src/Widgets/WifiInterface.vala

```diff
@@ -256,10 +259,17 @@ namespace Network {
                             wifi_dialog.run ();
                             wifi_dialog.destroy ();
                         } else {
-                            client.add_and_activate_connection (new NM.Connection (),
-                                                                wifi_device,
-                                                                row.ap.get_path (),
-                                                                finish_connection_cb);
+                            client.add_and_activate_connection_async.begin (NM.SimpleConnection.new (),
+                                                                            wifi_device,
+                                                                            row.ap.get_path (),
+                                                                            null,
+                                                                            (obj, res) => {
+                                                                                try {
+                                                                                    client.add_and_activate_connection_async.end (res);
+                                                                                } catch (Error error) {
+                                                                                    warning ("%s\n", error.message);
```

Comment:
`warning ()` takes care of properly formatting the string, so you can just do:
`warning (error.message)`.